File: leetcode2026/2026_01/leetcode_3047_largestSquareArea.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    long long largestSquareArea(vector<vector<int>>& bottomLeft, vector<vector<int>>& topRight) {
        int n = bottomLeft.size();
        long long res = 0;
        for (int i=0; i<n; i++){
            for (int j=i+1; j<n; j++){
                int x1 = max(bottomLeft[i][0], bottomLeft[j][0]);
                int y1 = max(bottomLeft[i][1], bottomLeft[j][1]);
                int x2 = min(topRight[i][0], topRight[j][0]);
                int y2 = min(topRight[i][1], topRight[j][1]);
                int x_length = max(x2-x1, 0), y_length = max(y2-y1, 0);
                int edge = min(x_length, y_length);
                res = max(res, 1ll*edge*edge);
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<vector<int>> bottomLeft = {{1,1},{2,2},{3,1}};
    vector<vector<int>> topRight = {{3,3},{4,4},{6,6}};
    cout << s.largestSquareArea(bottomLeft, topRight) << endl;
}

File: 04.Simulation模拟/leetcode_3047_largestSquareArea.py
from typing import List

class Solution:
    def largestSquareArea(self, bottomLeft: List[List[int]], topRight: List[List[int]]) -> int:
        def calc(l1, r1, l2, r2):
            length = 0
            if l1 <= l2 <= r1 <= r2:
                length = r1-l2
            elif l1 <= l2 <= r2 <= r1:
                length = r2-l2
            return length

        n = len(bottomLeft)
        res = 0
        for i in range(n):
            bl1 = bottomLeft[i]
            tr1 = topRight[i]
            for j in range(i+1, n):
                bl2 = bottomLeft[j]
                tr2 = topRight[j]
                
                x_length = max(calc(bl1[0], tr1[0], bl2[0], tr2[0]), 
                               calc(bl2[0], tr2[0], bl1[0], tr1[0]))
                
                y_length = max(calc(bl1[1], tr1[1], bl2[1], tr2[1]),
                               calc(bl2[1], tr2[1], bl1[1], tr1[1]))

                edge = min(x_length, y_length)
                res = max(res, edge*edge)
        return res

if __name__ == "__main__":
    s = Solution()
    bottomLeft = [[1,1],[2,2],[3,1]]
    topRight = [[3,3],[4,4],[6,6]]
    print(s.largestSquareArea(bottomLeft, topRight))
